File: java/src/main/java/com/promoengine/domain/PromotionId.java
// ABOUTME: Type-safe wrapper for promotion identifiers.
// ABOUTME: Ensures promotion IDs are non-empty strings.
package com.promoengine.domain;

public record PromotionId(String value) {

    public PromotionId {
        if (value == null || value.trim().isEmpty()) {
            throw new IllegalArgumentException("PromotionId cannot be null or empty");
        }
    }

    @Override
    public String toString() {
        return value;
    }
}

File: typescript/src/domain/PromotionId.ts
// ABOUTME: Promotion identifier with validation
// ABOUTME: Type-safe wrapper for promotion ID strings

export class PromotionId {
  private readonly _value: string;

  constructor(value: string) {
    if (!value || value.trim().length === 0) {
      throw new Error('PromotionId cannot be empty');
    }
    this._value = value.trim();
  }

  get value(): string {
    return this._value;
  }

  equals(other: PromotionId): boolean {
    return this._value === other._value;
  }

  toString(): string {
    return this._value;
  }
}
